core/variant/variant_internal.h

```diff
@@ -1125,40 +1066,23 @@ struct VariantInternalAccessor<Vector<Variant>> {
 	}
 };
 
+template <typename T, typename = void>
+struct VariantInitializer;
+
 template <typename T>
-struct VariantInitializer {
-};
-
-template <>
-struct VariantInitializer<bool> {
-	static _FORCE_INLINE_ void init(Variant *v) { VariantInternal::init_generic<bool>(v); }
-};
-
-#define INITIALIZER_INT(m_type)                        \
-	template <>                                        \
-	struct VariantInitializer<m_type> {                \
-		static _FORCE_INLINE_ void init(Variant *v) {  \
-			VariantInternal::init_generic<int64_t>(v); \
-		}                                              \
-	};
-
-INITIALIZER_INT(uint8_t)
-INITIALIZER_INT(int8_t)
-INITIALIZER_INT(uint16_t)
-INITIALIZER_INT(int16_t)
-INITIALIZER_INT(uint32_t)
-INITIALIZER_INT(int32_t)
-INITIALIZER_INT(uint64_t)
-INITIALIZER_INT(int64_t)
-INITIALIZER_INT(char32_t)
-INITIALIZER_INT(Error)
-INITIALIZER_INT(ObjectID)
-INITIALIZER_INT(Vector2::Axis)
-INITIALIZER_INT(Vector2i::Axis)
-INITIALIZER_INT(Vector3::Axis)
-INITIALIZER_INT(Vector3i::Axis)
-INITIALIZER_INT(Vector4::Axis)
-INITIALIZER_INT(Vector4i::Axis)
+struct VariantInitializer<T, std::enable_if_t<std::is_integral_v<T> || std::is_enum_v<T>>> {
+	static _FORCE_INLINE_ void init(Variant *v) { VariantInternal::init_generic<T>(v); }
+};
+
+template <typename T>
+struct VariantInitializer<BitField<T>, std::enable_if_t<std::is_enum_v<T>>> {
+	static _FORCE_INLINE_ void init(Variant *v) { VariantInternal::init_generic<BitField<T>>(v); }
+};
+
+template <>
+struct VariantInitializer<ObjectID> {
+	static _FORCE_INLINE_ void init(Variant *v) { VariantInternal::init_generic<ObjectID>(v); }
```

Comment:
Do these 3 cases have to be separate? I don't really see a difference.

Reply:
Now that you mention it, this whole section simplified drastically. The default case can just be `init_generic`!